### PR TITLE
#245: support the in operator for Circuit and CircuitBuilder

### PR DESCRIPTION
**Pull request by arcturusannamalai**
- implement __contains__ method

### CODE REVIEW

**Comment by rturrado** [comment]
@arcturusannamalai Can you please add some tests for this work? Thanks!

**Comment by arcturusannamalai** [comment]
need a bit more time; will do!

**Comment by arcturusannamalai** [comment]
done @rturrado

File: opensquirrel/circuit.py [hunk 1]
```diff
@@ -54,6 +54,9 @@ def __eq__(self, other: Any) -> bool:
             return False
         return self.register_manager == other.register_manager and self.ir == other.ir
 
+    def __contains__(self, qubit:Qubit) -> bool:
+        return any([qubit in stmt.get_qubit_operands()  for stmt in self.ir.statements])
```
Comment (rturrado) [inline opensquirrel/circuit.py:58]:
```suggestion
        return any([qubit in statement.get_qubit_operands()  for statement in self.ir.statements])
```

File: opensquirrel/circuit_builder.py [hunk 2]
```diff
@@ -61,6 +61,9 @@ def __init__(
         self.register_manager = RegisterManager(QubitRegister(qubit_register_size), BitRegister(bit_register_size))
         self.ir = IR()
 
+    def __contains__(self, qubit:Qubit) -> bool:
+        return any([qubit in stmt.get_qubit_operands()  for stmt in self.ir.statements])
```
Comment (rturrado) [inline opensquirrel/circuit_builder.py:65]:
```suggestion
        return any([qubit in statement.get_qubit_operands()  for statement in self.ir.statements])
```

File: test/test_circuit_builder.py [hunk 3]
```diff
@@ -49,6 +49,15 @@ def test_single_measure(self) -> None:
...
         ]
 
+    def test_contains(self) -> None:
+        builder = CircuitBuilder(1, 1)
+        builder.measure(Qubit(0), Bit(0))
+
+        circuit = builder.to_circuit()
+
+        assert not Qubit(1) in builder
+        assert Qubit(0) in circuit
```
Comment (rturrado) [inline test/test_circuit_builder.py:59]:
```suggestion
        assert Qubit(0) in circuit
        assert not Qubit(1) in builder
```

**Review by rturrado — CHANGES_REQUESTED** [review]
Just some minor aesthetic changes.

**Comment by rturrado** [comment]
@arcturusannamalai It would be worth checking where in the code it is now convenient to use the `in` operator.

Also, please make sure you run the requested tests and linters locally before committing. You can check what to run at the [Contributing](https://github.com/QuTech-Delft/OpenSquirrel/blob/develop/CONTRIBUTING.md#before-creating-the-pull-request) guide.

Thanks!

**Comment by arcturusannamalai** [comment]
@rturrado - please check my latest attempt :)

**Comment by rturrado** [comment]
> @rturrado - please check my latest attempt :)

I'm sorry @arcturusannamalai, but could you please explain what the latest commit has to do with the purpose of this PR?

**Comment by arcturusannamalai** [comment]
> > @rturrado - please check my latest attempt :)
> 
> I'm sorry @arcturusannamalai, but could you please explain what the latest commit has to do with the purpose of this PR?

I added code annotations etc. to pass the criteria in your project post-commit checks. It keeps logic and a new unit-test added for the test intact.

**Comment by rturrado** [comment]
> I added code annotations etc. to pass the criteria in your project post-commit checks. It keeps logic and a new unit-test added for the test intact.

Sorry, but that `force-pushed the ma-245 branch` contains a lot of changes I didn't understand.

Anyway, you keep missing running the static analysis tools.

**Pull request closed without merging**